llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4459,7 +4459,6 @@ static bool willGenerateVectors(VPlan &Plan, ElementCount VF,
       switch (R.getVPDefID()) {
       case VPDef::VPDerivedIVSC:
       case VPDef::VPScalarIVStepsSC:
-      case VPDef::VPScalarCastSC:
       case VPDef::VPReplicateSC:
       case VPDef::VPInstructionSC:
```

Comment:
VPInstructionSC now also holds for VPInstructionWithType recipes, instead of VPScalarCastSC.